src/shiv/pip.py

```diff
@@ -42,10 +43,16 @@ def install(args: List[str]) -> None:
     """
     with clean_pip_env():
 
+        # if being invoked as a pyz, we must ensure we have access to our own
+        # site-packages when subprocessing since there is no guarantee that pip
+        # will be available
+        subprocess_env = {**os.environ, **{"PYTHONPATH": sys.path[_first_sitedir_index() or 0]}}
```

Comment:
What if PYTHONPATH is set in os.environ? 
This will clobber it instead of extending it.

Also, would a simple `os.environ` update do instead of this dictionary expansion/creation?

Reply:
I suppose update would be fine here since it's during the lifecycle of the build process. I am always hesitant to permanently infect the environment, passing a copy of it to the subprocess feels a bit cleaner to me.

I agree that this is a bit "lazy" (clobbering PYTHONPATH instead of extending it, etc). Perhaps I can refactor `shiv.bootstrap._extend_python_path` to be re-usable here...